access-common/src/main/java/edu/unc/lib/dl/ui/util/DatastreamUtil.java

```diff
@@ -67,26 +55,35 @@ public String getDatastreamUrl(BriefObjectMetadata metadata, String datastreamNa
 
         StringBuilder url = new StringBuilder();
 
-        if (preferredDS.getExtension() != null) {
-            int extensionIndex = Arrays.binarySearch(new String[] { "doc", "docx", "htm", "html", "pdf", "ppt", "pptx",
-                    "rtf", "txt", "xls", "xlsx", "xml" }, preferredDS.getExtension());
-            if (extensionIndex >= 0) {
+        if (!isBlank(preferredDS.getExtension())) {
+            if (INDEXABLE_EXTENSIONS.contains(preferredDS.getExtension())) {
                 url.append("indexable");
             }
         }
 
         url.append("content/");
-        if (preferredDS.getOwner() == null) {
+        if (isBlank(preferredDS.getOwner())) {
             url.append(metadata.getId());
         } else {
             url.append(preferredDS.getOwner());
         }
-        if (!ContentModelHelper.Datastream.DATA_FILE.getName().equals(datastreamName)) {
+        if (!ORIGINAL_FILE.equals(datastreamName)) {
             url.append("/").append(preferredDS.getName());
```

Comment:
why is preferredDS.getName() being appended to the url rather than datastreamName?

Reply:
They are the same thing in this case, but I will change the conditional to check `preferredDS.getName()` for consistency